Src/IronPython/Runtime/TypecodeOps.cs

```diff
@@ -100,29 +119,29 @@ internal static class TypecodeOps {
                 case 'Q':
                     result = MemoryMarshal.Read<ulong>(bytes);
                     return true;
+                case 'P':
+                    result = MemoryMarshal.Read<UIntPtr>(bytes);
+                    return true;
                 default:
                     result = null;
                     return false;
             }
         }
 
-        public static bool TryGetBytes([NotNullWhen(true)]string? typecode, object obj, Span<byte> dest) {
-            if (string.IsNullOrEmpty(typecode) || typecode!.Length > 2 || (typecode.Length == 2 && typecode[0] != '@')) {
-                return false;
-            }
-            switch (typecode[typecode.Length - 1]) {
+        public static bool TryGetBytes(char typecode, object obj, Span<byte> dest) {
+            switch (typecode) {
                 case 'c':
-                    var cbyteVal = (byte)Convert.ToChar(obj);
-                    return MemoryMarshal.TryWrite(dest, ref cbyteVal);
+                    var bytecharVal = (byte)((Bytes)obj)[0];
```

Comment:
I know there's already a check in `MemoryView` to make sure it's a `Bytes` object, but the cast seems odd? Maybe it's fine, just caught my eye...

Reply:
It does seem odd. It is the consequence of the division of responsibilities between `memoryview` and `TypecodeOps` in general, and this method signature in particular. But it works. I saw little benefit rethinking/redesigning this in isolation from `struct`, so I've left it the way it is.